### PR TITLE
Added Missing Dependency Builtin Modules

### PR DESCRIPTION
**Pull request by antoine-unity**
### Purpose of this PR

Removing the module Particle System or IMGUI was causing compilation errors. Added both of those as dependencies. I validated that removing everything in the manifest file didn't cause any other errors.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+### Bug Fixes
+
+- Added Particle System and IMGUI modules as a dependency.
```
Comment (karl-) [inline CHANGELOG.md:12]: @lopezt-unity can we make particle system an optional dependency?

Comment (AverageComet250) [reply CHANGELOG.md:12]: Not really as without it you can't compile.

**Review by gytis-zasciurinskas — APPROVED** [review]
Checked the package compilation with 2019.4.20f1, 2020.2.3f1, 2021.1.0b5 and 2021.2.0a3 and it has compiled properly in each stream. I have also done some quick modeling with 21.2.0a3 specifically, to see if ProBuilder was functioning as expected. Haven't noticed any issues, so PR looks good!

File: CHANGELOG.md (hunk 1 above)
Comment (lopezt-unity) [reply CHANGELOG.md:12]: I added this as an optional dependency, it should be lighter.
Thanks Antoine for the help on this one.

**Review by lopezt-unity — APPROVED** [review]

**Review by karl- — APPROVED** [review]

**Pull request merged**